### DISCUSSION TITLE
Improve documentation regarding Enumerations

### DISCUSSION THREAD

**Discussion by lapidus** [schemaorg/schemaorg#3181] [Q&A] [answered]
The styleguide lacks information about Enumerations. 
https://schema.org/docs/styleguide.html

<img width="500" alt="image" src="https://user-images.githubusercontent.com/66601/190855715-2dd0136b-a221-43f9-95b8-9c0b56618951.png">

Here are some 'style topics' I'd love to see covered:

**1. Are there any naming conventions for Enumerations?**
Suffixes such as `Enumeration`, `Type`, `Option`, `Category`, `Value` are in use currently.

<img width="289" alt="image" src="https://user-images.githubusercontent.com/66601/190855709-4ef993cc-8317-4c50-9599-092e7a3ea76d.png">

**2. What are the criteria for turning something into an Enumeration?** 

Whereas there is a `DigitalPlatformEnumeration`, there's no `ComputerLanguageEnumeration`.
There is a `ComputerLanguage` is a class, but no `DigitalPlatform` class.
Would be nice if the documentation clarified the rationale here.

**3. Do Enumeration Members always use Class-like casing?** 

My understanding is that _Enumeration Members_ are instances (rdf:type) of the specific Enumeration. Yet, they use the same casing as classes, such as [`Online` ](https://schema.org/Online) or [AerobicActivity](https://schema.org/AerobicActivity).

Something like AerobicActivity seems like a good case for punning, where it occurs as **both** an instance of the [PhysicalActivityCategory](https://schema.org/PhysicalActivityCategory) and as a subclass of for example [PhysicalActivity](https://schema.org/PhysicalActivity). This seems analogous to the Wikidata example here: https://www.wikidata.org/wiki/Q1152037

Given that some Enumeration Members are unlikely (?) to be used as proper classes, what is the rationale for adopting the class casing? And more generally, would it be possible to provide some guidelines here, detailing typical characteristics that these Enumeration Members exhibit?

**Comment by danbri** [comment, marked as answer]
1. The best we could say is that there are a few patterns. Let's try to improve that!
2. Generally we're driven by applications that use the schemas, but we are now set up so that multiple related enumerations can be added alongside each other instead of being merged into a single list. See https://schema.org/EnergyEfficiencyEnumeration for example, it has EU and EnergyStar subtypes so their contents can remain more or less 1:1 with external standards.
4. They always look like types. This is not ideal but gives at least a clear way to tell if something is a property (it begins with a lowercase letter).

There are also cases like https://schema.org/CreditCard which have both an enumeration and an instance personality -

**Comment by rrlevering** [comment]
The one thing to mention about Google-specific interpretation right now (just for conversation about mass consumption that's not in the spec) is that we allow non-prefixed enumeration values as well (just like @types).  So we currently support `Monday`, `http://schema.org/Monday`, and `https://schema.org/Monday` as values for http://schema.org/dayOfWeek (but only when the enumeration is a known target of that predicate).
